cli/src/rich/mod.rs

```diff
@@ -358,13 +531,233 @@ fn draw_input(screen: &mut dyn Write, input: &Input) -> io::Result<()> {
         termion::cursor::Goto(4 + input.cursor as u16, input_row)
     )?;
 
+    if let Some(suggestion) = suggestion.map(|it| it.term.as_ref()) {
+        write!(
+            screen,
+            "{}{}{}{}{}{}{}",
+            termion::cursor::Save,
+            termion::color::Fg(termion::color::LightWhite),
+            termion::color::Bg(termion::color::Blue),
+            suggestion
+                .chars()
+                .skip(input.get_text().len())
+                .collect::<String>(),
+            termion::color::Reset.fg_str(),
+            termion::color::Reset.bg_str(),
+            termion::cursor::Restore,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn draw_autocomplete(
+    screen: &mut dyn Write,
+    autocomplete: Option<&Autocomplete>,
+) -> io::Result<()> {
+    let Some(autocomplete) = autocomplete else {
+        return Ok(());
+    };
+
+    let Some(term_width) = autocomplete.suggestions.iter().map(|i| i.term.len()).max() else {
+        return Ok(());
+    };
+    let Some(summary_width) = autocomplete
+        .suggestions
+        .iter()
+        .map(|i| i.summary.len())
+        .max()
+    else {
+        return Ok(());
+    };
+    let width = term_width + summary_width + 2;
+
+    let (_, term_height) = termion::terminal_size().unwrap();
+    let start_row = term_height - 2 - autocomplete.len() as u16;
+
+    for (pos, suggestion) in autocomplete.suggestions.iter().enumerate() {
+        let padding =
+            String::from_utf8(vec![
+                b' ';
+                width - suggestion.term.len() - suggestion.summary.len()
+            ])
+            .unwrap();
```

Comment:
The math here could underflow if the suggestion is very long/the terminal is very narrow. Also, I guess you're not actually allocating twice here because `String::from_utf8` works in place, but it's still more convoluted than it needs to be because you can collect directly into a String.

How about

```suggestion
        let padding: String = std::iter::repeat(' ')
            .take(
                width
                    .checked_sub(suggestion.term.len() + suggestion.summary.len())
                    .unwrap_or(1)
                )
            )
            .collect();
```

Reply:
Way nicer, thanks!